### DISCUSSION TITLE
Welcome to megabouts Discussions!

### DISCUSSION THREAD

**Discussion by AlexandreLaborde** [orger-lab/megabouts#4] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by HrmnKnng** [comment]
Hello Orger lab, 

congrats on building this immensely useful tool! I have been finding my way around the code to adjust and apply it to my datasets and the tutorials on the webpage have been a great help on getting through the pipeline. One thing that would be a massive help to anyone working with Megabouts would be a clear list describing what is exactly contained in each of the pipeline's output variables. 
Additionally, it would be a great to have the option to input data from multiple animals recordings, this would prevent having to split up multi track files from SLEAP into single tracks, running them through megabouts separately and then combine them again afterwards.

Finally, I would like to ask if/how preprocessing deals with missing frames in the input file. The cvs files that SLEAP produces only contains data for the frames it managed to create an instance for and no tracking is perfect so when animals "evade" tracking, those frames are completely absent in the cvs file (I believe they're accounted for in the hdf5 files). I pre-preprocess my SLEAP cvs files by adding rows of NaNs for missing frames to be ensure that time stays linear and to be able to fairly downsample my data before inputting it to the megabouts pipeline. 

Again, great work on this project!

Cheers,
Harmen,

McLean lab,
Centre for Discovery Brain Science,
University of Edinburgh.

**Reply by AlexandreLaborde** [reply]
Hi Harmen,

Thank you so much for your kind words and thoughtful feedback! I’m thrilled to hear that you’re finding the tool useful and that the tutorials have been helpful in navigating the pipeline. It’s always great to hear how the tool is being applied to real-world datasets—thank you for sharing your experience 😊

I just added the 4 points you mentioned to our issues and I will start working on them soon.

Regarding your second point about handling multi-animal recordings, I’d be happy to implement this feature but I don't have any data in this format. Can you please send me some example data?

Thanks again for your detailed feedback and for taking the time to share your thoughts. It’s incredibly valuable as we continue to improve Megabouts. Let me know if there’s anything else I can do to support your work!

Best regards,
**Alexandre Laborde**
_Computer Vision Developer @ Digital Surgery Lab
Champalimaud Clinical Centre_

**Reply by HrmnKnng** [reply]
Hi Alexandre,

great to hear that my input got logged as issues, very happy to give you some early adapter inputs.

For the multiple animal tracking data, I have attached a folder with some data files for you. If anything is unclear about the data or you need more/other files, please let me know.
[multi_animal_files_for_OrgerLab.zip](https://github.com/user-attachments/files/18766256/multi_animal_files_for_OrgerLab.zip)

Another question that popped up about the software is what happens to bouts that can't be fitted into any of the 13 predefined behavioral categories? Are they excluded from the output or are they assigned to their closest matching category but with a very low probability/certainty? Is there a way of flagging these kind of bouts? I thinking of less frequently occurring behaviors like struggling or for example a user case where one would want to use megabouts to run a locomotion analysis on mutants with a possible locomotor phenotype.

All the best,
Harmen

**Comment by AlexandreLaborde** [comment]
Issues created related to this #8 #9 #10 #11
